### PR TITLE
Logging 2

### PR DESCRIPTION
**Pull request by mcasimir**
Removes dependencies on winston, cleaner logging.

### CODE REVIEW

File: src/DefaultLogger.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      info:     2,
+      verbose:  3,
+      debug:    4,
+      silly:    5
+    };
+
+    this.level = options.level || 'info';
+  }
+
+  getCurrentPriority() {
```
Comment (raoulus) [inline src/DefaultLogger.js:21]: You might wanna call it getCurrentLogLevel

File: src/Logger.js [hunk 2]
```diff
@@ -1,33 +1,36 @@
...
-  setLevel: function(level) {
-    this.winston.level = level;
+  constructor(impl) {
+    this.impl = impl || new DefaultLogger();
   }
-};
 
-module.exports = {
+  setImpl(impl) {
+    this.impl = impl;
```
Comment (raoulus) [inline src/Logger.js]: what about renaming `impl` to `instance`?

Comment (raoulus) [inline src/Logger.js]: As it's a singleton I'd rename it to instance

File: src/Logger.js [hunk 3]
```diff
@@ -1,33 +1,36 @@
...
+  }
 
-  getGlobalLogger: function() {
-    return globalLogger;
-  },
+  getImpl() {
```
Comment (mcasimir) [inline src/Logger.js]: That can be confusing, remove

File: src/Logger.js [hunk 4]
```diff
@@ -1,33 +1,36 @@
...
-
-  setLevel: function(level) {
-    this.winston.level = level;
+  constructor(impl) {
+    this.impl = impl || new DefaultLogger();
   }
-};
 
-module.exports = {
+  setImpl(impl) {
```
Comment (mcasimir) [inline src/Logger.js]: rename to setLogger

**Pull request merged**